compiler/src/dotty/tools/dotc/core/Mode.scala

```diff
@@ -70,26 +70,29 @@ object Mode {
   /** We are currently unpickling Scala2 info */
   val Scala2Unpickling = newMode(13, "Scala2Unpickling")
 
+  /** We are currently unpickling from Java 8 or higher */
+  val Java8Unpickling = newMode(14, "Java8Unpickling")
```

Comment:
it would be nice to leave a note that it is used as heuristic to find scala 2.12